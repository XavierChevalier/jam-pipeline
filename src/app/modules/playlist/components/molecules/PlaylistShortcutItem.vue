<script setup lang="ts">
  defineProps({
    cover: {
      type: String,
      required: true,
    },
    title: {
      type: String,
      required: true,
    },
    description: {
      type: String,
      required: true,
    },
  })
</script>

<template>
  <div class="flex flex-col min-w-[150px] w-[150px]">
    <img class="rounded" :src="cover" :alt="`Cover of ${title}`" />
    <span class="font-bold mt-2">{{ title }}</span>
    <span class="text-sm">{{ description }}</span>
  </div>
</template>
