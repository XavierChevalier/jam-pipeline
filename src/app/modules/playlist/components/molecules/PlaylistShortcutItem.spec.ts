import * as Stories from '@/app/modules/playlist/components/molecules/PlaylistShortcutItem.stories'
import { Album } from '@/app/modules/playlist/components/molecules/PlaylistShortcutItem.stories'
import { StorybookTestsGenerator } from '@/tests/tests-generators/StorybookTestsGenerator'
import { shallowMount } from '@vue/test-utils'

describe('PlaylistShortcutItem', () => {
  const component = StorybookTestsGenerator.fromStoriesExports(Stories)
    .snapshotEachStories()
    .property<string>('cover', (property) => {
      property.itShouldBeDefined().itShouldBeRequired().itShouldBeTypeOf(String)
    })
    .property<string>('title', (property) => {
      property.itShouldBeDefined().itShouldBeRequired().itShouldBeTypeOf(String)
    })
    .property<string>('description', (property) => {
      property.itShouldBeDefined().itShouldBeRequired().itShouldBeTypeOf(String)
    })
    .getComponent()

  const wrapper = shallowMount(component, { props: Album.args })
  it('should display the cover image', () => {
    const coverElement = wrapper.find('img')
    expect(coverElement.attributes('src')).toContain(wrapper.props('cover'))
    expect(coverElement.attributes('alt')).toContain(wrapper.props('title'))
  })

  it('should display the title text', () => {
    expect(wrapper.text()).toContain(wrapper.props('title'))
  })

  it('should display the description text', () => {
    expect(wrapper.text()).toContain(wrapper.props('description'))
  })
})
