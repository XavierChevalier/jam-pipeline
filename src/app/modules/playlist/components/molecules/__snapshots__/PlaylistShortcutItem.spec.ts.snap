// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaylistShortcutItem stories snapshots [snapshot] should render the "Album" component story correctly 1`] = `
<div
  class="flex flex-col min-w-[150px] w-[150px]"
>
  <img
    alt="Cover of Polyphia"
    class="rounded"
    src="https://picsum.photos/160"
  />
  <span
    class="font-bold mt-2"
  >
    Polyphia
  </span>
  <span
    class="text-sm"
  >
    New levels new devils
  </span>
</div>
`;
